ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/parser.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright © Aptos Foundation
+
+use crate::{
+    db::upsert_uris,
+    models::{NFTMetadataCrawlerURIs, NFTMetadataCrawlerURIsQuery},
+};
+use diesel::{
+    r2d2::{ConnectionManager, PooledConnection},
+    PgConnection,
+};
+use google_cloud_auth::token_source::TokenSource;
+use image::{ImageBuffer, ImageFormat};
+use nft_metadata_crawler_utils::{
+    gcs::{write_image_to_gcs, write_json_to_gcs},
+    NFTMetadataCrawlerEntry,
+};
+use serde_json::Value;
+use tracing::{error, info};
+
+// Stuct that represents a parser for a single entry from queue
+pub struct Parser<'a> {
+    pub entry: NFTMetadataCrawlerEntry,
+    model: NFTMetadataCrawlerURIs,
```

Comment:
If `Parser` represents a single entity in the queue, can we rename it to something like `ParsingEntity/Unity/Entry` or `ProcessingEntity/Unity/Entry`